client/config/config.go

```diff
@@ -0,0 +1,152 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	tmcli "github.com/tendermint/tendermint/libs/cli"
+)
+
+var ErrWrongNumberOfArgs = fmt.Errorf("wrong number of arguments")
+
+type ClientConfig struct {
+	ChainID        string `mapstructure:"chain-id"`
+	KeyringBackend string `mapstructure:"keyring-backend"`
+	Output         string `mapstructure:"output"`
+	Node           string `mapstructure:"node"`
+	BroadcastMode  string `mapstructure:"broadcast-mode"`
+	Trace          bool   `mapstructure:"trace"`
+}
+
+func DefaultClientConfig() *ClientConfig {
+	return &ClientConfig{
+		ChainID:        "",
+		KeyringBackend: "os",
+		Output:         "text",
+		Node:           "tcp://localhost:26657",
+		BroadcastMode:  "sync",
+		Trace:          false,
+	}
+}
+
+// Cmd returns a CLI command to interactively create an application CLI
+// config file.
+func Cmd(defaultCLIHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config <key> [value]",
+		Short: "Create or query an application CLI configuration file",
+		RunE:  runConfigCmd,
+		Args:  cobra.RangeArgs(0, 2),
+	}
+
+	cmd.Flags().String(flags.FlagHome, defaultCLIHome,
+		"set client's home directory for configuration")
+	//	cmd.Flags().Bool(flagGet, false,
+	//		"print configuration value or its default if unset")
+	return cmd
+}
+
+func runConfigCmd(cmd *cobra.Command, args []string) error {
+
+	v := viper.New()
+
+	cfgPath, err := ensureCfgPath(v.GetString(flags.FlagHome))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to make config path: %v\n", err)
+		return err
+	}
+
+	cliConfig, err := getClientConfig(cfgPath, v)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to get client config: %v\n", err)
+		return err
+	}
+
+	switch len(args) {
+	case 0:
+		// print all client config fields to sdt out
+		// TODO implement method to print out all client config fields
+		fmt.Println(cliConfig.ChainID)
+		fmt.Println(cliConfig.KeyringBackend)
+		fmt.Println(cliConfig.Output)
+		fmt.Println(cliConfig.Node)
+		fmt.Println(cliConfig.BroadcastMode)
+		fmt.Println(cliConfig.Trace)
+
+	case 1:
+		// it's a get
+		// TODO implement method for get
+		// should i implement getters here?
+		key := args[0]
+		switch key {
+		case flags.FlagChainID:
+			fmt.Println(cliConfig.ChainID)
+		case flags.FlagKeyringBackend:
+			fmt.Println(cliConfig.KeyringBackend)
+		case tmcli.OutputFlag:
+			fmt.Println(cliConfig.Output)
+		case flags.FlagNode:
+			fmt.Println(cliConfig.Node)
+		case flags.FlagBroadcastMode:
+			fmt.Println(cliConfig.BroadcastMode)
+		case "trace":
+			fmt.Println(cliConfig.Trace)
+		default:
+			err := errUnknownConfigKey(key)
+			fmt.Fprintf(os.Stderr, "Unable to get the value for the key: %v, error:  %v\n", key, err)
+			return err
+		}
+
+	case 2:
+		// it's set
+		// TODO impement method for set
+		// TODO implement setters
```

Comment:
Yes, maybe setters would make sense, because we need to perform some basic validation (for example, keyring-backend can only take one of 5-6 values) for the fields